core/testing.go

```diff
@@ -20,38 +23,44 @@ func MockConfig(t *testing.T) *Config {
 }
 
 // StartMockNode starts a mock Core node background process and returns it.
-func StartMockNode() *node.Node {
-	app := kvstore.NewApplication()
-	app.RetainBlocks = 10
+func StartMockNode(app types.Application) *node.Node {
 	return rpctest.StartTendermint(app, rpctest.SuppressStdout, rpctest.RecreateConfig)
 }
 
 func EphemeralMockEmbeddedClient(t *testing.T) Client {
-	nd := StartMockNode()
+	nd := StartMockNode(CreateKvStore(defaultRetainBlocks))
 	t.Cleanup(func() {
 		nd.Stop() //nolint:errcheck
 		rpctest.StopTendermint(nd)
 	})
 	return NewEmbeddedFromNode(nd)
 }
 
+// CreateKvStore creates a simple kv store app and gives the user
+// ability to set desired amount of blocks to be retained.
+func CreateKvStore(retainBlocks int64) *kvstore.Application {
+	app := kvstore.NewApplication()
+	app.RetainBlocks = retainBlocks
+	return app
+}
```

Comment:
optional nit: you can remove requirements of the retainBlocks param and anyway change it on the returned value directly.

Reply:
Ok